full-moon/src/ast/compound.rs

```diff
@@ -0,0 +1,154 @@
+use crate::ast::{Expression, Var};
+use crate::tokenizer::{Symbol, TokenReference};
+use derive_more::Display;
+use full_moon_derive::{Node, Visit};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Display, PartialEq, Eq, Node, Visit)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[non_exhaustive]
+#[allow(missing_docs)]
+#[display("{_0}")]
+/// Compound operators, such as X += Y or X -= Y
+pub enum CompoundOp {
+    PlusEqual(TokenReference),
+    MinusEqual(TokenReference),
+    StarEqual(TokenReference),
+    SlashEqual(TokenReference),
+    CaretEqual(TokenReference),
```

Comment:
IMO for this and the checks you should do `#[cfg(any(feature = "luau", feature = "cfxlua"))]`--it makes it clear when adding more what's actually supported by Lua itself (nothing) and what isn't.